### PR TITLE
DOC: Add experimental note to DatetimeArray and TimedeltaArray

### LINKED ISSUE #24870: IntervalArray not documented in What's New for 0.24.0rc1
#### Problem description

The new type `IntervalArray` is in the API reference, but is not mentioned in What's New in the docs for v0.24.0rc1 .  The existence of the new type is probably something people would like to know about.

http://pandas.pydata.org/pandas-docs/version/0.24.0rc1/api/arrays.html#interval-data has it listed there, but in http://pandas.pydata.org/pandas-docs/version/0.24.0rc1/whatsnew/v0.24.0.html it appears in an example, and that there is a new method for the type.

Not sure if @TomAugspurger or @jschendel should look at that.

### PR DESCRIPTION
**Pull request by TomAugspurger**
DOC: Mention PeriodArray and IntervalArray

Closes #24870

### CODE REVIEW

**Comment by jreback** [comment]
is it worth putting a similar warning in whatsnew / docs-proper as well (maybe in timeseries.rst & timedelta.rst top-section)?

**Comment by TomAugspurger** [comment]
Wasn't sure. Neither of those have sections for DatetimeArray or TimedletaArray specifically. The only place is in `api/arrays.rst`.

**Comment by jreback** [comment]
@TomAugspurger yeah right. Maybe we you add that (certainly not necessary for now though)

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -107,6 +107,29 @@ def wrapper(self, other):
 
 
 class TimedeltaArray(dtl.DatetimeLikeArrayMixin, dtl.TimelikeOps):
+    """
+    Pandas ExtensionArray for timedelta data.
+
+    .. versionadded :: 0.24.0
```
Comment (jschendel) [inline pandas/core/arrays/timedeltas.py]: I think the space between "versionadded" and "::" should be removed.

**Review by jschendel — APPROVED** [review]
Thanks! LGTM aside from one small comment.

**Comment by jorisvandenbossche** [comment]
I would add similar notices to IntegerArray, and to the whatsnew file (but can also be in separate PR)

**Comment by TomAugspurger** [comment]
@jorisvandenbossche is 
https://github.com/pandas-dev/pandas/pull/24882/commits/e65fa9077ad3991965795499990e2d8bc77f1f2e what you had in mind for IntegerArray?

**Comment by jorisvandenbossche** [comment]
> @jorisvandenbossche is
e65fa90 what you had in mind for IntegerArray?

yep! can you add the same in the integer_na.rst ?

**Comment by TomAugspurger** [comment]
Done in 
https://github.com/pandas-dev/pandas/pull/24882/commits/5bec9e47a1ead6ad2ed07a4072b15a32a88e8186

**Pull request merged**